internal/security/fileprovider/provider_test.go

```diff
@@ -0,0 +1,420 @@
+	mockAuthTokenLoader.AssertExpectations(t)
+	mockSecretStoreClient.AssertExpectations(t)
+	assert.Equal(t, expectedTokenFile("myservice"), service1Buffer.Bytes())
+}
+
+// TestNoDefaultsCustomTokenParameters
+func TestNoDefaultsCustomTokenParameters(t *testing.T) {
+	// Arrange
+	privilegedTokenPath := "/dummy/privileged/token.json"
+	configFile := "token-config.json"
+	outputDir := "/outputdir"
+	outputFilename := "secrets-token.json"
+
+	mockLogger := logger.MockLogger{}
+
+	mockFileIoPerformer := &MockFileIoPerformer{}
+	expectedService1Dir := filepath.Join(outputDir, "myservice")
+	expectedService1File := filepath.Join(expectedService1Dir, outputFilename)
+	service1Buffer := new(bytes.Buffer)
+	mockFileIoPerformer.On("MkdirAll", expectedService1Dir, os.FileMode(0700)).Return(nil)
+	mockFileIoPerformer.On("OpenFileReader", configFile, os.O_RDONLY, os.FileMode(0400)).Return(strings.NewReader(`{"myservice":{"custom_token_parameters":{"key1":"value1"}}}`), nil)
+	mockFileIoPerformer.On("OpenFileWriter", expectedService1File, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, os.FileMode(0600)).Return(&writeCloserBuffer{service1Buffer}, nil)
+
+	mockAuthTokenLoader := &MockAuthTokenLoader{}
+	mockAuthTokenLoader.On("Load", privilegedTokenPath).Return("fake-priv-token", nil)
+
+	expectedService1Policy := "{}"
+	expectedService1Parameters := make(map[string]interface{})
+	expectedService1Parameters["key1"] = "value1"
+	expectedService1Parameters["meta"] = makeMetaServiceName("myservice")["meta"]
+	mockSecretStoreClient := &MockSecretStoreClient{}
+	mockSecretStoreClient.On("InstallPolicy", "fake-priv-token", "edgex-service-myservice", expectedService1Policy).Return(http.StatusNoContent, nil)
+	mockSecretStoreClient.On("CreateToken", "fake-priv-token", expectedService1Parameters, mock.Anything).
+		Run(func(args mock.Arguments) {
+			setCreateTokenResponse("myservice", args.Get(2).(*interface{}))
+		}).
+		Return(http.StatusOK, nil)
+
+	p := NewTokenProvider(mockLogger, mockFileIoPerformer, mockAuthTokenLoader, mockSecretStoreClient)
+	p.SetConfiguration(secretstoreclient.SecretServiceInfo{}, config.TokenFileProviderInfo{
+		PrivilegedTokenPath: privilegedTokenPath,
+		ConfigFile:          configFile,
+		OutputDir:           outputDir,
+		OutputFilename:      outputFilename,
+	})
+
+	// Act
+	err := p.Run()
+
+	// Assert
+	// - {OutputDir}/myservice/{OutputFilename} w/proper contents
+	// - Correct token parameters for myservice
+	// - All other expectations met
+	assert.NoError(t, err)
+	mockFileIoPerformer.AssertExpectations(t)
+	mockAuthTokenLoader.AssertExpectations(t)
+	mockSecretStoreClient.AssertExpectations(t)
+	assert.Equal(t, expectedTokenFile("myservice"), service1Buffer.Bytes())
+}
+
+// TestTokenUsingDefaults
+func TestTokenUsingDefaults(t *testing.T) {
+	// Arrange
+	privilegedTokenPath := "/dummy/privileged/token.json"
+	configFile := "token-config.json"
+	outputDir := "/outputdir"
+	outputFilename := "secrets-token.json"
```

Comment:
I saw this repeats in several places in the test code, should consider to make it as a constant and use it to reduce the maintenance if wanna change in future.

Reply:
Moved these to a const block at the top.